draco/core/io.py

```diff
@@ -982,6 +982,131 @@ def process(self):
+
+        if not self.output_name.endswith(".zarr.zip"):
+            raise ConfigError("File output name must end in `.zarr.zip`.")
+
+        # Trim off the .zip suffix and fix the file format
+        self.output_name = self.output_name[:-4]
+        self.output_format = fileformats.Zarr
+        self.save = True
+
+        # Call the baseclass to determine which ranks will do the work
+        super().setup()
+
+    # Override next as we don't want the usual mechanism
+    def next(self, container: memh5.BasicCont) -> ZarrZipHandle:
+        """Take a container and save it out as a .zarr.zip file.
+
+        Parameters
+        ----------
+        container
+            Container to save out.
+
+        Returns
+        -------
+        handle
+            A handle to use to determine if the job has successfully completed. This
+            should be given to the `WaitZarrZip` task.
+        """
+
+        outfile = self._save_output(container)
+        dest_file = outfile + ".zip"
+        self.comm.Barrier()
+
+        bg_process = None
+
+        host_rank_to_use = self._operation_counter % self._num_hosts
+
+        if self._host_rank == host_rank_to_use:
+
+            self.log.info(f"Starting background job to zip {outfile}")
+
+            # Run 7z to zip up the file
+            dest_file = outfile + ".zip"
+            src_dir = outfile + "/."
+            command = f"{self._path_7z} a -tzip -mx=0 {dest_file} {src_dir}"
+
+            # If we are to remove the file get the background job to do it immediately
+            # after zipping succeeds
+            if self.remove:
+                command += " && rm -r {outfile}"
+
+            bg_process = subprocess.Popen(
+                command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+            )
+
+        # Increment the global operations counter
+        self.__class__._operation_counter += 1
+
+        return ZarrZipHandle(dest_file, bg_process)
+
+
+class WaitZarrZip(task.MPILoggedTask):
+    """Collect Zarr-zipping jobs and wait for them to complete."""
+
+    _handles: Optional[List[ZarrZipHandle]] = None
+
+    def next(self, handle: ZarrZipHandle):
+        """Receive the handles to wait on.
+
+        Parameters
+        ----------
+        handle
+            The handle to wait on.
+        """
+
+        if self._handles is None:
+            self._handles = []
+
+        self._handles.append(handle)
+
+    def finish(self):
+        """Wait for all Zarr zipping jobs to complete."""
+
+        for h in self._handles:
+
+            self.log.debug(f"Waiting on job processing {h.filename}")
+
+            if h.handle is not None:
+                returncode = h.handle.wait()
+
+                if returncode != 0 or not os.path.exists(h.filename):
+                    self.log.debug("Error occurred while zipping. Debug logs follow...")
+                    self.log.debug(f"stdout={h.handle.stdout}")
```

Comment:
If we're including this sort of debugging, I think this would be useful
```suggestion
                    self.log.debug(f"returncode={returncode}")
                    self.log.debug(f"stdout={h.handle.stdout}")
```